src/main.rs

```diff
@@ -308,6 +316,98 @@ fn configure_sdk(args: &CliArgs) -> Result<()> {
     Ok(())
 }
 
+fn sign_fragmented(
+    manifest: &mut Manifest,
+    signer: &dyn Signer,
+    init_pattern: &PathBuf,
+    frag_pattern: &PathBuf,
+    output_path: &PathBuf,
+) -> Result<()> {
+    // search folders for init segments
+    let ip = init_pattern.to_str().ok_or(c2pa::Error::OtherError(
+        "could not parse source pattern".into(),
+    ))?;
+    let inits = glob::glob(ip)
+        .map_err(|_e| c2pa::Error::OtherError("could not process glob pattern".into()))?;
+    let mut count = 0;
+    for init in inits {
+        match init {
+            Ok(p) => {
+                let mut fragments = Vec::new();
+                let init_dir = p.parent().unwrap();
+                let seg_glob = init_dir.join(frag_pattern); // segment match pattern
+
+                // grab the fragments that go with this init segment
+                for seg in glob::glob(seg_glob.to_str().unwrap()).unwrap().flatten() {
+                    fragments.push(seg);
+                }
+
+                println!("Adding manifest to: {:?}", p);
+                let new_output_path = output_path.join(init_dir.file_name().unwrap());
+                manifest.embed_to_bmff_fragmented(&p, &fragments, &new_output_path, signer)?;
+
+                count += 1;
+            }
+            Err(_) => bail!("bad path to init segment"),
+        }
+    }
+    if count == 0 {
+        println!("No files matching pattern: {}", ip);
+    }
+    Ok(())
+}
+
+fn verify_fragmented(
+    init_pattern: &PathBuf,
+    frag_pattern: &PathBuf,
+) -> Result<Vec<ManifestStore>> {
+    let mut stores = Vec::new();
+
+    let ip = init_pattern.to_str().ok_or(c2pa::Error::OtherError(
+        "could not parse source pattern".into(),
+    ))?;
+    let inits = glob::glob(ip)
+        .map_err(|_e| c2pa::Error::OtherError("could not process glob pattern".into()))?;
+    let mut count = 0;
+
+    // search folders for init segments
+    for init in inits {
+        match init {
+            Ok(p) => {
+                let mut fragments = Vec::new();
+                let init_dir = p.parent().unwrap();
+                let seg_glob = init_dir.join(frag_pattern); // segment match pattern
+
+                // grab the fragments that go with this init segment
+                for seg in glob::glob(seg_glob.to_str().unwrap()).unwrap().flatten() {
+                    fragments.push(seg);
+                }
+
+                println!("Verifying manifest: {:?}", p);
+                let store = ManifestStore::from_fragments(p, &fragments, true)?;
+                if let Some(vs) = store.validation_status() {
+                    if let Some(e) =  vs.iter().find(|v| !v.passed()) 
+                    {
+                        println!("Error validating segments: {:?}", e);
```

Comment:
generally with errors we use eprintln or bail. I take it you want this to be valid since we return Ok(stores) below. maybe at least eprinln! since it it an error?